Enforce test style on e2e_node and e2e_kubeadm tests

odinuge opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

This enforces the policy no more "Expect(err).NotTo(HaveOccurred())" on the e2e_node and e2e_kubeadm folders. Previously it was just enforced on the e2e-tests.


**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #34059 (enforced by the script), there are still ~1500 `framework.ExpectNoError(err)`s without an assertion message tho.

Fixes #77103
/priority backlog


**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

odinuge commented:
/assign @oomichi

oomichi commented:
@odinuge big thanks for doing this :-)

/ok-to-test

oomichi commented:
This is necessary to be updated for passing pull-kubernetes-bazel-build job by fixing the following failures:
```
/b/f/w/test/e2e_node/benchmark_util.go:36:2: imported and not used: "k8s.io/kubernetes/vendor/github.com/onsi/gomega"
/b/f/w/test/e2e_node/pods_container_manager_test.go:31:2: imported and not used: "k8s.io/kubernetes/vendor/github.com/onsi/gomega"
/b/f/w/test/e2e_node/resource_usage_test.go:33:2: imported and not used: "k8s.io/kubernetes/vendor/github.com/onsi/gomega"
/b/f/w/test/e2e_node/runtime_conformance_test.go:34:2: imported and not used: "k8s.io/kubernetes/vendor/github.com/onsi/gomega"
/b/f/w/test/e2e_node/volume_manager_test.go:30:2: imported and not used: "k8s.io/kubernetes/vendor/github.com/onsi/gomega"
```

odinuge committed:
Cleanup e2e_node test style

Remove the use of  Except(err).NotTo(HaveOccured()), and switch to using
framework ramework.ExpectNoError(err)

odinuge committed:
Run verify-test-code on all e2e tests

oomichi commented:
Thanks for updating,

/lgtm

oomichi commented:
/approve

odinuge commented:
/retest

Had to fix some conflicting files @oomichi 😄

oomichi commented:
@odinuge OK, I will check this again after passing all test jobs.
Thanks for updating :-)

oomichi commented:
/test pull-kubernetes-integration

odinuge commented:
/test pull-kubernetes-integration

oomichi commented:
/test pull-kubernetes-bazel-test

oomichi commented:
/lgtm

odinuge commented:
/assign @vishh

BenTheElder commented on hack/verify-test-code.sh:
```diff
@@ -21,7 +21,7 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 
 # NOTE: This checks e2e test code without the e2e framework which contains Expect().To(HaveOccurred())
-mapfile -t all_e2e_files < <(find test/e2e -name '*.go' | grep -v 'test/e2e/framework/')
+mapfile -t all_e2e_files < <(find test/e2e{,_node,_kubeadm} -name '*.go' | grep -v 'test/e2e/framework/')
```
NOTE: see previous discussions in this repo: we don't use mapfile because it doesn't exist on macOS. there are examples of alternatives eleswhere in hack/ such as the verify-shellcheck script (read loop)

BenTheElder replied:
this particular change looks fine, but the original implementation probably should not have introduced mapfile usage

oomichi replied:
@BenTheElder I see, thanks for your comment. I will follow the mapfile issue with another PR with some digging.

odinuge replied:
There is a tool inside `hack/lib/util.sh` called `kube::util::read-array` that works the same way as `mapfile -t`, and works in bash3 :smile: @oomichi @BenTheElder But yeah, that can be fixed in another PR.

https://github.com/kubernetes/kubernetes/blob/b3a73f761f5b67f783d40a8b89dce47c5a015909/hack/lib/util.sh#L715

BenTheElder replied:
@oomichi @odinuge thanks! that's exactly the thing to do, let's do that in another PR. feel free to ping me on that one for approval :sweat_smile:

oomichi replied:
https://github.com/kubernetes/kubernetes/pull/79186 is the follow-up PR.

BenTheElder reviewed (approved):
/lgtm
/approve
we should follow up on the one comment, but it wasn't introduced in this PR..
